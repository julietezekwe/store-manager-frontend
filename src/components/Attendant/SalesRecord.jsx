import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getPersonalSales } from '../../actions/salesAction';

export class SalesRecord extends Component {
  async componentDidMount() {
    await this.props.getPersonalSales();
  }

  render() {
    const { mySales } = this.props.sales;
    return (
      <React.Fragment>
        <div className="col-8">
          <div className="center">
            <div className="col-3 " id="totalProducts">
              Total Products Sold:{' '}
              {mySales.records && mySales.records.totalProducts}
            </div>
            <div className="col-4" id="totalPrice">
              Amount Realized: &#8358;{' '}
              {mySales.records && mySales.records.amount}
            </div>
            <div className="col-3" id="totalSales">
              Number of sales: {mySales.records && mySales.records.totalSales}
            </div>
          </div>

          <div className="card col-11">
            <table className="table">
              <thead>
                <tr>
                  <th>Product Id</th>
                  <th>Product Name</th>
                  <th>Quantity</th>
                  <th>Price</th>
                  <th>Seller</th>
                  <th>Date</th>
                </tr>
              </thead>
              <tbody>
                {mySales.saleDetail
                  && mySales.saleDetail.map((sale, i) => {
                    const date = new Date(sale.created_at);
                    const dateString = new Date(
                      date.getTime() - date.getTimezoneOffset() * 60000,
                    )
                      .toISOString()
                      .split('T')[0];

                    return (
                      <tr key={i}>
                        <td>{sale.productid}</td>
                        <td>{sale.productname}</td>
                        <td>{sale.quantity}</td>
                        <td>{sale.price}</td>
                        <td>{this.props.user.name}</td>
                        <td>{dateString}</td>
                      </tr>
                    );
                  })}
              </tbody>
            </table>
          </div>
        </div>
      </React.Fragment>
    );
  }
}
const mapStateToProps = state => ({
  sales: state.sales,
  user: state.login.user,
});
export default connect(
  mapStateToProps,
  { getPersonalSales },
)(SalesRecord);
