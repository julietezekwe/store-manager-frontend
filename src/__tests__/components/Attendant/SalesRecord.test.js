/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { SalesRecord } from '../../../components/Attendant/SalesRecord.jsx';

let wrapper;
const props = {
  getPersonalSales: () => Promise.resolve(),
  sales: {
    mySales: {
      records: {
        totalProducts: 10,
        amount: 12099,
        totalSales: 4,
      },
      saleDetail: [{
        productid: 1,
        productname: 'shoes',
        description: 'fine',
        quantity: 10,
        price: 1200,
        created_at: new Date(),
      }],
    },
  },
  user: {
    id: 3,
    role: 'attendant',
    email: 'chidimma@yahoo.com',
    name: 'chidimma',
    username: 'chidimma',
  },
};
beforeEach(() => {
  wrapper = mount(<SalesRecord {...props} />);
});
describe('SalesRecord Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
