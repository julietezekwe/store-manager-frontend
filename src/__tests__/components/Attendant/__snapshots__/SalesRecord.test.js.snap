// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SalesRecord Component should match snapshot 1`] = `
<SalesRecord
  getPersonalSales={[Function]}
  sales={
    Object {
      "mySales": Object {
        "records": Object {
          "amount": 12099,
          "totalProducts": 10,
          "totalSales": 4,
        },
        "saleDetail": Array [
          Object {
            "created_at": 2019-02-23T18:18:22.556Z,
            "description": "fine",
            "price": 1200,
            "productid": 1,
            "productname": "shoes",
            "quantity": 10,
          },
        ],
      },
    }
  }
  user={
    Object {
      "email": "chidimma@yahoo.com",
      "id": 3,
      "name": "chidimma",
      "role": "attendant",
      "username": "chidimma",
    }
  }
>
  <div
    className="col-8"
  >
    <div
      className="center"
    >
      <div
        className="col-3 "
        id="totalProducts"
      >
        Total Products Sold:
         
        10
      </div>
      <div
        className="col-4"
        id="totalPrice"
      >
        Amount Realized: â‚¦
         
        12099
      </div>
      <div
        className="col-3"
        id="totalSales"
      >
        Number of sales: 
        4
      </div>
    </div>
    <div
      className="card col-11"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Product Id
            </th>
            <th>
              Product Name
            </th>
            <th>
              Quantity
            </th>
            <th>
              Price
            </th>
            <th>
              Seller
            </th>
            <th>
              Date
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="0"
          >
            <td>
              1
            </td>
            <td>
              shoes
            </td>
            <td>
              10
            </td>
            <td>
              1200
            </td>
            <td>
              chidimma
            </td>
            <td>
              2019-02-23
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</SalesRecord>
`;
